odoo/modules/module.py

```diff
@@ -311,74 +341,75 @@ def get_module_root(path):
-
-        info['version'] = adapt_version(info['version'])
-        return info
-
-    _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
-    return {}
+
+    if not manifest_file:
+        _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
+        return {}
+
+    manifest = copy.deepcopy(_DEFAULT_MANIFEST)
+    manifest['icon'] = get_module_icon(module)
+
+    f = tools.file_open(manifest_file, mode='rb')
+    try:
+        manifest.update(ast.literal_eval(pycompat.to_text(f.read())))
+    finally:
+        f.close()
+
+    if not manifest.get('description'):
+        readme_path = [opj(mod_path, x) for x in README
+                       if os.path.isfile(opj(mod_path, x))]
+        if readme_path:
+            with tools.file_open(readme_path[0]) as fd:
+                manifest['description'] = fd.read()
+
+    if not manifest.get('license'):
+        manifest['license'] = 'LGPL-3'
+        _logger.warning("Missing `license` key in manifest for %r, defaulting to LGPL-3", module)
+
+    # auto_install is either `False` (by default) in which case the module
+    # is opt-in, either a list of dependencies in which case the module is
+    # automatically installed if all dependencies are (special case: [] to
+    # always install the module), either `True` to auto-install the module
+    # in case all dependencies declared in `depends` are installed.
+    if isinstance(manifest['auto_install'], collections.abc.Iterable):
+        manifest['auto_install'] = set(manifest['auto_install'])
+        non_dependencies = manifest['auto_install'].difference(manifest['depends'])
+        assert not non_dependencies,\
+            "auto_install triggers must be dependencies, found " \
+            "non-dependencies [%s] for module %s" % (
+                ', '.join(non_dependencies), module
+            )
+    elif manifest['auto_install']:
+        manifest['auto_install'] = set(manifest['depends'])
+
+    manifest['version'] = adapt_version(manifest['version'])
+    manifest['addons_path'] = realpath(opj(mod_path, os.pardir))
+
+    return manifest
+
+@functools.lru_cache(maxsize=None)
+def load_manifest(module, mod_path=None):
+    """
+    Get the manifest dictionnary from the file-system.
+
+    :param str module: The name of the module (sale, purchase, ...).
+    :param Optionnal[str] mod_path: The optionnal path to the module on
+        the file-system, it is determined by scanning the addons-paths
+        when the argument is not set.
+    :return dict: The module manifest as a dictionnary or an empty one
+        when the manifest was not found.
+    """
+    return _load_manifest(module, mod_path)
+
+def load_information_from_description_file(module, mod_path=None):
+    warnings.warn(
+        'load_information_from_description_file() is a deprecated '
+        'alias to load_manifest()', DeprecationWarning, stacklevel=2)
+    return load_manifest(module, mod_path)
```

Comment:
do you think we could get rid of this entirely? it's also a technical API that only the framework is supposed to be using.

Reply:
While I agree with you about removing `read_manifest`, this one is actually often used, even *outside* of framework-like stuff. I'm not saying they are legit utilizing it but they are, often to get that `assets_path` key. Let's just keep the deprecation until saas-16.1. I know you hate when we bloat your logs with deprecation warnings stuff. This one is reasonable.